docs(CSS): Clarify style() container query support, simplify examples (#24481)
mdn/content: files/en-us/web/css/@container/index.md

### Values

- `<container-condition>`: A set of features that are evaluated against the query container when the container changes size.
  Styles defined in the `<stylesheet>` are applied if the condition is true.
  More details on the syntax are covered in the following sections.

> - `<stylesheet>`: A set of CSS declarations.

Author: Good idea, applied in 6b8794e71358eadd9d07c814ccd6c85b40f9435c

Details about usage and naming restrictions are described in the {{cssxref("container-name")}} page.

### Style container queries

Currently only size-based container queries have browser support.
It is expected that `style()` container queries will allow checking the {{cssxref("computed_value")}} of a container in future:

```css example-bad
/* Not yet supported */
@container style(color: blue;) {
  /* <stylesheet> */
}
```

> 

Reviewer: How about removing this section altogether for now and documenting the style container queries only after they're supported in browsers?

Perhaps a note can be added along with the value description for `<container-condition>` that only size-based container queries are supported at the moment.

Author: Yeah let's remove, I agree.

## Examples

### Setting styles based on a container's size

Consider the following example of a card component with a title and some text:

```html
<div class="post">
  <div class="card">
    <h2>Card title</h2>
    <p>Card content</p>
  </div>
</div>
```

> A container context can be created using the `container-type` property, in this case using the `inline-size` value on the `.post` element.
> You can then use the `@container` at-rule to apply styles to an element with the `.card` class if the element is in a container that's narrower than `650px`.

Reviewer: Simplifying the second sentence for readability:

```suggestion
A container context can be created using the `container-type` property, in this case using the `inline-size` value on the `.post` class.
You can then use the `@container` at-rule to apply styles to the element with the `.card` class in a container that's narrower than `650px`.
```

```js hidden
const post = document.querySelector(".post");
const span = document.createElement("span");
span.innerHTML = ".post width: " + post.clientWidth + "px";
post.parentNode.insertBefore(span, post.nextSibling);
// update on resize
window.addEventListener("resize", () => {
  span.innerHTML = ".post width: " + post.clientWidth + "px";
});
```

```css hidden
span {
  display: block;
  text-align: center;
}
.card {
  margin: 10px;
  border: 2px dotted;
  font-size: 1.5em;
}
.post {
  border: 2px solid;
}
```

```css
/* A container context based on inline size */
.post {
  container-type: inline-size;
}

/* Apply styles if the container is narrower than 650px */
@container (width < 650px) {
  .card {
    width: 50%;
    background-color: gray;
    font-size: 1em;
  }
}
```

{{EmbedLiveSample("Setting_styles_based_on_a_container's_size", "100%", 230)}}

> ### Named container contexts

Reviewer: ```suggestion
### Creating named container contexts
```

### Basic example

> Given the following HTML example which is a card component with a title and some text:

Reviewer: How about:

```suggestion
Consider the following example of a card component with a title and some text:
```

> A container context can be created using the `container-type` property:

Reviewer: ```suggestion
A container context can be created using the `container-type` property, as shown below:
```

```css
.post {
  container-type: inline-size;
}

> ```

Reviewer: This does not seem to be used in this example. Is this needed here?
Maybe it can be removed since it will be demo'd in the next example section on "Named container contexts".

Author: You do need it, I've updated the example in cb9f2cde8b7dd34f1e991b604f851d3adf4cb860 to include it so it's more explicit.

> The following example will apply styles to the `.card` element if it's in a container that is wider than 400px:

Reviewer: Edit suggestion to indicate that this is still part of the same example:

```suggestion
You can then use the `@container` at-rule to apply styles to an element with the `.card` class if the element is in a container that's wider than `400px`.
```

```css
@container (min-width: 400px) {
  .card {
    font-size: 1.5em;
  }
}
```

### Named container contexts

> Given the following HTML example which is a card component with a title, and some text:

Author: ```suggestion
Given the following HTML example which is a card component with a title and some text:
```

```html
<div class="post">
  <div class="card">
    <h2>Card title</h2>
    <p>Card content</p>
  </div>
</div>
```

> First create a container context using `container-type`, add the `container-name` property.

Reviewer: ```suggestion
First, create a container context using the `container-type` and `container-name` properties.
```

The shorthand syntax for this declaration is described in the {{cssxref("container")}} page.

```css
.post {
  container-type: inline-size;
  container-name: summary;
}
```

> You can then target that container by adding the name to the container query:

Reviewer: ```suggestion
Next, target that container by adding the name to the container query:
```

It's not possible to target multiple containers in a single container query.
It is possible to nest container queries which has the same effect.

> The following query evaluates to true and applies a style to child elements if the container named `card` is wider than 400px and it has an ancestor container that has a width greater than 800px:

Reviewer: ```suggestion
The following query evaluates to true and applies the declared style if the container named `summary` is wider than `400px` and has an ancestor container wider than `800px`:
```

would it be the child element? I guess it the style would be applied to whatever selector is declared in `<stylesheet>`.

Author: good 👍🏻
